var/spack/repos/builtin/packages/pika/package.py

```diff
@@ -146,5 +146,9 @@ def cmake_args(self):
             args += [self.define("CMAKE_CXX_COMPILER", self.spec["hip"].hipcc)]
             if self.spec.satisfies("^cmake@3.21.0:3.21.2"):
                 args += [self.define("__skip_rocmclang", True)]
+        if "@0.8: +rocm" in self.spec:
```

Comment:
We generally recommend using `self.spec.satisfies()` when checking version ranges (https://spack.readthedocs.io/en/latest/packaging_guide.html#testing-spec-constraints), which this includes.

```suggestion
        if self.spec.satisfies("@0.8: +rocm"):
```

Reply:
Thanks for pointing that out! The `if "@:0.7 ..."` line earlier should presumably also be changed in that case?

To be honest I can't quite grok what the difference between the two is based on the documentation you linked. It says "The difference between the two functions is that `satisfies()` tests whether spec constraints overlap at all, while `in` tests whether a spec or any of its dependencies satisfy the provided spec". "spec constraints overlap at all" and "satisfy the provided spec" sounds similar-ish. The only meaningful distinction I see there is that the latter says "a spec or any of its dependencies", but I don't quite understand the relevance of it. Is there another place explaining the difference in more detail? More importantly, is there _any_ reason to prefer `in` (besides that it reads nicely)?

Reply:
I just updated the package to use `satisfies` but I'm still curious about the difference between the two :)